docs/cudf/source/index.rst

```diff
@@ -14,6 +14,7 @@ the details of CUDA programming.
    :caption: Contents:
 
    user_guide/index
+   developer_guide/index
    api_docs/index
```

Comment:
I would copy the style of pandas and put the Developer Guide _after_ the API docs. People will consult the API docs far more frequently.
https://pandas.pydata.org/docs/

(NumPy, scikit-learn, Arrow, and other packages also put developer info near the bottom / end of their docs.)